File: src/dataset/c/collatz_conjecture_Exercism_m3g4d1v3r/claude/ClaudeSonnet4_collatz_conjecture_v1.c
#include "collatz_conjecture.h"

int steps(int start) {
    if (start <= 0) return ERROR_VALUE;
    
    int step_counter = 0;
    
    while (start != 1) {
        if (start & 1) {  // odd number
            start = (start << 1) + start + 1;  // 3*start + 1
        } else {  // even number
            start >>= 1;  // start / 2
        }
        step_counter++;
    }
    
    return step_counter;
}

File: src/dataset/python/collatz-conjecture/claude/ClaudeSonnet4_collatz-conjecture_v1.py
def steps(number):
    if number < 1:
        raise ValueError("Only positive integers are allowed")
    
    count = 0
    while number != 1:
        if number & 1:  # odd check using bitwise AND
            number = number * 3 + 1
        else:  # even
            number >>= 1  # bit shift right (divide by 2)
        count += 1
    return count